mobile/App.js

```diff
@@ -6,24 +6,89 @@
  * @flow strict-local
  */
 
-import React, {useState} from 'react';
+import React, {useState, useRef, useEffect} from 'react';
 import {SafeAreaView} from 'react-native-safe-area-context';
+import AsyncStorage from '@react-native-async-storage/async-storage';
+
 import HomeView from './src/views/HomeView';
 import VendorView from './src/views/VendorView';
 import RegistrationView from './src/views/RegistrationView';
 import {NavigationContainer} from '@react-navigation/native';
 import {createBottomTabNavigator} from '@react-navigation/bottom-tabs';
 import ScreenOption from './src/utilites/ScreenOption';
-import {globalStyleSheet} from './src/utilites/Theme';
 import MyICardView from './src/views/MyICardView';
-
+import { AppState } from "react-native";
 import AuthContext from './src/context/AuthContext';
+import { storeDate } from './src/utilites/StoreDate';
+import { storeUser } from './src/utilites/StoreUser';
+import { storeDateKey, storedUserDataKey, monthInMilliseconds } from './src/utilites/GlobalConstants';
+
 
 const Tab = createBottomTabNavigator();
 
 const App = () => {
   const [user, setUser] = useState(null);
+  const [check, setCheck] = useState(false);
   const value = {user, setUser};
+  const appState = useRef(AppState.currentState);
+  
+  const logoutCheckOnStartupAndOnForeground = async(userData = null,startup = false) => {
+    
+    try{
+        const lastDate = parseInt(await AsyncStorage.getItem(storeDateKey));    
+        if(lastDate != null){
```

Comment:
Sorry something I forgot as well is this if does not check if lastDate is NaN (Not a Number) as well. We should make it something like:
`if(lastDate != null && !isNaN(lastDate))`